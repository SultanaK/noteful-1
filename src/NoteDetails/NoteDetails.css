.note__details{
    padding: 20px 20px 40px 20px;
    margin: 10px 0px 10px 20px;
    width: 50%;
    height: 100%;
    box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
    border-radius: 10px;
    background: #fffefb;
}

p.note__error{
    margin-left: 40px;
}