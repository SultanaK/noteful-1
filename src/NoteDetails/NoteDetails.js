import React from 'react';
import STORE from '../dummy-store.js';
import './NoteDetails.css';

export default function NoteDetails(props){
    let note = (STORE.notes.find(note => note.id === props.match.params.noteId));

    return(<div className="note__details">
            <h3>{note.name}</h3>
            <p>Date Modified: {note.modified}</p>
            <p>{note.content}</p>
        </div>
    );
}