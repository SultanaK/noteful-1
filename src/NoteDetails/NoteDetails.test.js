import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { BrowserRouter } from 'react-router-dom';
import NoteDetails from './NoteDetails';

describe('Note Details Page', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter>
          <NoteDetails notes={sampleNotes} noteId={sampleNotes[0].id}/>
        </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders UI correctly', () => {
        const tree = renderer
          .create(<BrowserRouter><NoteDetails notes={sampleNotes} noteId={sampleNotes[0].id}/></BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

const sampleNotes = [
    {
        "id": "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Cats",
        "modified": "2018-08-15T23:00:00.000Z",
        "folderId": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "content": "Id vitae qui. Et ullam et eos. Suscipit perferendis et dolor non deserunt et et inventore exercitationem. Consequatur nihil ut impedit numquam ducimus. Culpa aut repellat sunt. Quas sapiente modi ea quia qui nihil maiores.\n \rIllo enim omnis illum dolorem. Voluptatem sequi pariatur id voluptatem quos. Ratione debitis laboriosam soluta optio. Eum nobis culpa sed quasi commodi sunt nesciunt doloribus accusantium.\n \rQuam cum quidem voluptas repudiandae tenetur error aut. Sit et explicabo voluptas dolore eum saepe omnis libero aut. Sit tempore voluptas voluptas ipsa ut explicabo ipsum et. Officiis minima tempora omnis consequatur est facilis eligendi nobis vel. Aliquam sapiente veniam corporis quas est omnis quis dolor dolor. Ut ducimus nobis perspiciatis debitis neque."
    },
    {
        "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Pigs",
        "modified": "2018-03-01T00:00:00.000Z",
        "folderId": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "content": "Quaerat qui similique nihil. Aut eveniet architecto id. Fugit aut ab quam. Excepturi officia et voluptatem vel rerum voluptates quas ullam facere.\n \rDebitis quis amet. Pariatur repellendus occaecati nulla. Ducimus sit quia non asperiores. Hic unde autem.\n \rDignissimos cupiditate fuga fuga accusantium et distinctio ad aut. Nihil atque reprehenderit nihil eius in. Rem illo sint suscipit quaerat ex incidunt et ut quae."
    }
];