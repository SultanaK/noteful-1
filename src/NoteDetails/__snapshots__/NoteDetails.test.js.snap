// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Details Page renders UI correctly 1`] = `
<div
  className="note__details"
>
  <h3>
    Cats
  </h3>
  <p>
    Date Modified: 
    2018-08-15T23:00:00.000Z
  </p>
  <p>
    Id vitae qui. Et ullam et eos. Suscipit perferendis et dolor non deserunt et et inventore exercitationem. Consequatur nihil ut impedit numquam ducimus. Culpa aut repellat sunt. Quas sapiente modi ea quia qui nihil maiores.
 
Illo enim omnis illum dolorem. Voluptatem sequi pariatur id voluptatem quos. Ratione debitis laboriosam soluta optio. Eum nobis culpa sed quasi commodi sunt nesciunt doloribus accusantium.
 
Quam cum quidem voluptas repudiandae tenetur error aut. Sit et explicabo voluptas dolore eum saepe omnis libero aut. Sit tempore voluptas voluptas ipsa ut explicabo ipsum et. Officiis minima tempora omnis consequatur est facilis eligendi nobis vel. Aliquam sapiente veniam corporis quas est omnis quis dolor dolor. Ut ducimus nobis perspiciatis debitis neque.
  </p>
</div>
`;
